### PR TITLE
Issue 5083: add description field to v1 link props

### LINKED ISSUE #5083: Add new field `description` into /v1/link/props API
Part of #5072
At this moment we have only 2 link props: `max_bandwidth` and `cost`. We need to add a new one - `description`

To store ticket reference (which is described in #5072) we need to create a new field in ISL model.
It includes:
1. Add `description` field into LinkProps DB model
2. Add `description` field into Isl DB model
3. Create a DB migration for these 2 fields.
4. Add description field into following APIs:
`GET /v1/links`
`DELETE /v1/links`
Add description prop handling into following APIs:
`GET /v1/link/props`
`PUT /v1/link/props`
`DELETE /v1/link/props`

### PR DESCRIPTION
**Pull request by DejfCold**
Closes #5083

### CODE REVIEW

File: docker/db-migration/migrations/025-add-description-prop.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+databaseChangeLog:
+  - changeSet:
+      id: tag
+      author: dstudeny
+      changes:
+        - tagDatabase:
+            tag: 024-add-description-prop
```
Comment (niksv) [inline docker/db-migration/migrations/025-add-description-prop.yaml:7]: tag is wrong. it must be -25

Comment (IvanChupin) [reply docker/db-migration/migrations/025-add-description-prop.yaml]: fixed

File: docker/db-migration/migrations/025-add-description-prop.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+databaseChangeLog:
+  - changeSet:
+      id: tag
+      author: dstudeny
+      changes:
+        - tagDatabase:
+            tag: 025-add-description-prop
```
Comment (dmitrii-beliakov) [inline docker/db-migration/migrations/025-add-description-prop.yaml:7]: There will be a conflict with [025-optimise-switch-connected-device-class.yaml](https://github.com/telstra/open-kilda/pull/5108/files)

File: src-java/northbound-service/northbound/src/test/java/org/openkilda/northbound/converter/LinkMapperTest.java [hunk 3]
```diff
@@ -76,7 +76,8 @@ public void testMapResponse() {
                 false,
                 true,
                 BFD_SESSION_STATUS,
-                PACKET_ID);
+                PACKET_ID,
+                null);
```
Comment (dmitrii-beliakov) [inline src-java/northbound-service/northbound/src/test/java/org/openkilda/northbound/converter/LinkMapperTest.java:80]: Why not to test the description mapping?

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkPropertiesSpec.groovy [hunk 4]
```diff
@@ -174,6 +179,10 @@ class LinkPropertiesSpec extends HealthCheckSpecification {
         islUtils.getIslInfo(updatedLinks, isl).get().maxBandwidth == maxBandwidthValue.toInteger()
         islUtils.getIslInfo(updatedLinks, isl.reversed).get().maxBandwidth == maxBandwidthValue.toInteger()
 
+        and: "Description on forward and reverse ISLs us really updated as well"
```
Comment (dmitrii-beliakov) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkPropertiesSpec.groovy:182]: us? a typo

**Review by dmitrii-beliakov — COMMENTED** [review]
Please fix a typo and consider to add tests where appropriate

File: src-java/northbound-service/northbound/src/test/java/org/openkilda/northbound/converter/LinkMapperTest.java (hunk 3 above)
Comment (niksv) [reply src-java/northbound-service/northbound/src/test/java/org/openkilda/northbound/converter/LinkMapperTest.java:80]: +1

File: docker/db-migration/migrations/025-add-description-prop.yaml (hunk 2 above)
Comment (durgakako) [reply docker/db-migration/migrations/025-add-description-prop.yaml]: Updated.

File: src-java/northbound-service/northbound/src/test/java/org/openkilda/northbound/converter/LinkMapperTest.java (hunk 3 above)
Comment (durgakako) [reply src-java/northbound-service/northbound/src/test/java/org/openkilda/northbound/converter/LinkMapperTest.java]: Updated.

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkPropertiesSpec.groovy (hunk 4 above)
Comment (durgakako) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkPropertiesSpec.groovy]: Updated

**Review by niksv — APPROVED** [review]

**Pull request merged**